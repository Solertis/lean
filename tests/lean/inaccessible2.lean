inductive imf {A B : Sort*} (f : A → B) : B → Sort*
| mk : ∀ (a : A), imf (f a)

definition inv_1 {A B : Sort*} (f : A → B) : ∀ (b : B), imf f b → A
| .(f .a) (imf.mk .f a)  := a  -- Error inaccessible annotation inside inaccessible annotation

definition inv_2 {A B : Sort*} (f : A → B) : ∀ (b : B), imf f b → A
| .(f a) (let x := (imf.mk .f a) in x)  := a  -- Error invalid occurrence of 'let' expression

definition inv_3 {A B : Sort*} (f : A → B) : ∀ (b : B), imf f b → A
| .(f a) ((λ (x : imf f b), x) (imf.mk .f a)) := a  -- Error invalid occurrence of 'lambda' expression

definition symm {A : Sort*} : ∀ a b : A, a = b → b = a
| .a .a (eq.refl a) := rfl -- Error `a` in eq.refl must be marked as inaccessible since it is an inductive datatype parameter
