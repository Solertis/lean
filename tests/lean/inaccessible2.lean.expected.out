inaccessible2.lean:5:7: error: invalid occurrence of 'inaccessible' annotation, it must only occur in patterns
inaccessible2.lean:8:10: error: invalid pattern, must be an application, constant, variable, type ascription or inaccessible term
inaccessible2.lean:11:11: error: invalid pattern, must be an application, constant, variable, type ascription or inaccessible term
inaccessible2.lean:14:9: error: invalid pattern, in a constructor application, the parameters of the inductive datatype must be marked as inaccessible
